scripts/shellcheck.sh

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env bash
+
+CYAN="\e[36m"
+YELLOW="\e[33m"
+BLUE="\e[34m"
+RED="\e[31m"
+GREEN="\e[32m"
+RST_COLOR="\e[0m"
+
+ROOT="$(cd "$(dirname "$0")"/.. && pwd)"
+SHELLCHECK_REPORT="${ROOT}/shellcheck-report.txt"
+START_TIME=$(date "+%Y-%m-%d %H:%M:%S")
+
+show_usage() {
+    cat << EOF
+Usage: $0 [options]
+
+Options:
+    -h, --help               print this help and exit
+    -e, --exclude-code       exclude specific error code, can be repeated
+    -S, --severity=LEVEL     set severity level (info, warning, error)
+    -v, --view               view shellcheck report
+
+Usage: ./scripts/shellcheck.sh [-e|--exclude-code=CODE] [-S|--severity=LEVEL] [-v|--view]
+
+example: ./scripts/shellcheck.sh -e SC1091 -e SC1090 -S warning -v
+
+EOF
+}
+
+parse_cli_args() {
+    echo; echo "Command line arguments: $0 $*"; echo
```

Comment:
Are we planning to include this?

Reply:
Optional, I could go either way on it

Reply:
Personally, I'd view this as unhelpful noise unless guarded by a debugging option.

Reply:
ACK, I'll put it under the help section so it only prints if triggered.

Reply:
I do not think this is helpful to have in the help output by-default.